### PR TITLE
Citizen application editor base

### PR DESCRIPTION
**Pull request by jaahon**
#### Summary

Add a base for application editor code.
Also, modify new citizen applications endpoint by not returning applications that are hidden from the guardian from the api.

### CODE REVIEW

File: frontend/packages/citizen-frontend/src/applications/api.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+export async function getApplication(
+  applicationId: string
+): Promise<Result<Application>> {
+  try {
+    const { data } = await client.get<JsonOf<Application>>(
+      `/citizen/applications/${applicationId}`
+    )
+    return Success.of(deserializeApplication(data))
+  } catch (e) {
+    return Failure.fromError(e)
```
Comment (reynders) [inline frontend/packages/citizen-frontend/src/applications/api.ts:16]: Should this error be logged?

File: frontend/packages/citizen-frontend/src/applications/api.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        : null
+    },
+    preferences: {
+      ...json.form.preferences,
+      preferredStartDate: LocalDate.parseNullableIso(
+        json.form.preferences.preferredStartDate
+      )
+    }
+  },
+  createdDate: json.createdDate ? new Date(json.createdDate) : null,
```
Comment (reynders) [inline frontend/packages/citizen-frontend/src/applications/api.ts:54]: This and modifiedDate is always set

File: frontend/packages/citizen-frontend/src/applications/editor/AdditionalDetailsSection.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// SPDX-FileCopyrightText: 2017-2020 City of Espoo
```
Comment (reynders) [inline frontend/packages/citizen-frontend/src/applications/editor/AdditionalDetailsSection.tsx:1]: 2021

File: frontend/packages/citizen-frontend/src/applications/editor/Heading.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import { H1, P } from '@evaka/lib-components/src/typography'
+import { useTranslation } from '~localization'
+
+export default React.memo(function Heading({ type }: { type: 'daycare' }) {
+  const t = useTranslation()
+  const infoParagraphs = t.applications.editor.heading.info[type]
+
+  return (
+    <ContentArea opaque paddingVertical="L">
+      <H1 noMargin>{t.applications.editor.heading.title[type]}</H1>
```
Comment (reynders) [inline frontend/packages/citizen-frontend/src/applications/editor/Heading.tsx:19]: Application list localization shares some of these values, lets combine these later

**Review by reynders — APPROVED** [review]

File: frontend/packages/citizen-frontend/src/applications/api.ts (hunk 2 above)
Comment (jaahon) [reply frontend/packages/citizen-frontend/src/applications/api.ts:54]: But they are also nullable in backend.

File: frontend/packages/citizen-frontend/src/applications/api.ts (hunk 1 above)
Comment (jaahon) [reply frontend/packages/citizen-frontend/src/applications/api.ts:16]: I don't know. We never log these.

File: frontend/packages/citizen-frontend/src/applications/api.ts (hunk 2 above)
Comment (reynders) [reply frontend/packages/citizen-frontend/src/applications/api.ts:54]: That should be fixed at some point in the backend then, this is ok for now

**Pull request merged**